src/processor.rs

```diff
@@ -1,69 +1,53 @@
-use tokio::io::AsyncWriteExt;
-use tokio::net::TcpStream;
+use anyhow::Result;
 // pyO3 module
 use crate::types::PyFunction;
 use pyo3::prelude::*;
 
-use hyper::{Body, Method, Request, Response, StatusCode};
+use hyper::{Body, Response, StatusCode};
 
 // Handle message fetches the response function
 // function is the response function fetched from the router
 // tokio task is spawned depending on the type of function fetched (Sync/Async)
 
 pub async fn handle_request(
-    function: Option<PyFunction>,
-    status_code: u16,
+    function: PyFunction,
 ) -> Result<Response<Body>, hyper::Error> {
-    match &function {
-        None => {
-            if status_code == 404 {
-                let mut not_found = Response::default();
-                *not_found.status_mut() = StatusCode::NOT_FOUND;
-                return Ok(not_found);
-            } else if status_code == 400 {
-                let mut not_found = Response::default();
-                *not_found.status_mut() = StatusCode::NOT_FOUND;
-                return Ok(not_found);
-            }
+    let contents = match execute_function(function).await {
+        Ok(res) => res,
+        Err(err) => {
+            println!("Error: {:?}", err);
+            let mut not_found = Response::default();
+            *not_found.status_mut() = StatusCode::INTERNAL_SERVER_ERROR;
+            return Ok(not_found);
         }
-        Some(_) => {}
-    }
+    };
 
-    let function = function.unwrap();
+    Ok(Response::new(Body::from(contents)))
+}
 
-    let contents = match function {
+#[inline]
```

Comment:
One more question, what does an `#[inline]` do in rust?

Reply:
It hints to the compiler that the assembly should be placed inside the caller function, this can make calling functions faster especially if in a commonly called function. The slight cost is a marginally larger binary.